src/platform/forms/address/helpers.js

```diff
@@ -128,39 +67,49 @@ export function formatAddress(address) {
 
   const {
     addressLine1,
-    addressLine3,
     addressLine2,
+    addressLine3,
+    addressType,
     city,
+    countryCodeIso3,
     countryName,
     internationalPostalCode,
-    militaryPostOfficeTypeCode,
-    militaryStateCode,
     province,
     stateCode,
-    type,
     zipCode,
   } = address;
 
-  const country = type === ADDRESS_TYPES.international ? countryName : '';
   let cityStateZip = '';
 
+  const displayCountry = countries.find(
+    country => country.countryCodeISO3 === countryCodeIso3,
+  );
+
+  const displayCountryName = displayCountry?.countryName;
+
+  // Only show country when ADDRESS_TYPES.international
+  const country =
+    addressType === ADDRESS_TYPES.international
+      ? countryName || displayCountryName
+      : '';
+
   const street =
     [addressLine1, addressLine2, addressLine3]
       .filter(item => item)
       .join(', ') || '';
 
-  switch (type) {
+  // Do not look up stateCode when ADDRESS_TYPES.military
```

Comment:
Should this comment say to `only look up for ADDRESS_TYPES.domestic`?

Reply:
That's much clearer :smile: